docs/adrs/adr1_value_object.md

```diff
@@ -0,0 +1,32 @@
+# How to implement value objects in our domains
+
+While modelling domains we use concept of value object. The issue is how to achieve model of value object described in DDD.
+
+## Status
+
+To be implemented (04.02.21)
+
+## Context
+
+We see two options: Base abstract class and record.
+
+We used base class before records were introduced. The issue with base class is that it forced developer to implement `GetAttributesToIncludeInEqualityCheck` method in every implemented ValueObject and always contained all the properties that were defined.
```

Comment:
```suggestion
We used base class before records were introduced. The issue with the base class is that it forces developers to implement `GetAttributesToIncludeInEqualityCheck` method in every implemented `ValueObject` and always contained all the properties that were defined.
```

Also, what do you mean by "and always contained all the properties that were defined."?

Reply:
I was thinking about full property definition not like the short one like in records, but I guess it is not that crucial here, so I will remove this bit.